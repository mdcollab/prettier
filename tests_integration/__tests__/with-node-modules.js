"use strict";

const runPrettier = require("../runPrettier");

test("ignores node_modules by default", () => {
  const result = runPrettier("cli/with-node-modules", ["**/*.js", "-l"]);

  expect(result.stdout).toMatchSnapshot();
  expect(result.status).toEqual(1);
});

test("ignores node_modules by with ./**/*.js", () => {
  const result = runPrettier("cli/with-node-modules", ["./**/*.js", "-l"]);

  expect(result.stdout).toMatchSnapshot();
  expect(result.status).toEqual(1);
});

test("doesn't ignore node_modules with --with-node-modules flag", () => {
  const result = runPrettier("cli/with-node-modules", [
    "**/*.js",
    "-l",
    "--with-node-modules"
  ]);

  expect(result.stdout).toMatchSnapshot();
  expect(result.status).toEqual(1);
});

test("ignores node_modules by default for file list", () => {
  const result = runPrettier("cli/with-node-modules", [
    "node_modules/node-module.js",
    "not_node_modules/file.js",
    "regular-module.js",
    "-l"
  ]);

  expect(result.stdout).toMatchSnapshot();
  expect(result.status).toEqual(1);
});

test("doesn't ignore node_modules with --with-node-modules flag for file list", () => {
  const result = runPrettier("cli/with-node-modules", [
    "node_modules/node-module.js",
    "not_node_modules/file.js",
    "regular-module.js",
    "-l",
    "--with-node-modules"
  ]);

  expect(result.stdout).toMatchSnapshot();
  expect(result.status).toEqual(1);
});
