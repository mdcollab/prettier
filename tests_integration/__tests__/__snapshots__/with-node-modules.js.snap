// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`doesn't ignore node_modules with --with-node-modules flag 1`] = `
"node_modules/node-module.js
not_node_modules/file.js
regular-module.js
"
`;

exports[`doesn't ignore node_modules with --with-node-modules flag for file list 1`] = `
"node_modules/node-module.js
not_node_modules/file.js
regular-module.js
"
`;

exports[`ignores node_modules by default 1`] = `
"not_node_modules/file.js
regular-module.js
"
`;

exports[`ignores node_modules by default for file list 1`] = `
"not_node_modules/file.js
regular-module.js
"
`;

exports[`ignores node_modules by with ./**/*.js 1`] = `
"./not_node_modules/file.js
./regular-module.js
"
`;
